deepreg/dataset/loader/h5_loader.py

```diff
@@ -143,4 +143,10 @@ def get_num_images(self) -> int:
     def close(self):
         """Close opened h5 file handles."""
         for f in self.h5_files.values():
-            f.close()
+            if f.__bool__():
+                # close file if it is open
+                try:
+                    f.close()
+                except RuntimeError:
+                    # to ignore RuntimeError: Can't increment id ref count
```

Comment:
Is this the best handling case for a runtime error?

Reply:
i got strange error that only occurs in CI not locally, I have no other methods :(

Reply:
Can we not at least log a warning saying that file did not close instead? Catching an error and not notifying the user isn't ideal.

Reply:
Yes, I am a bit worried about the way we handle this, it could come back to bite back...? Better to investigate the root cause and solve (in a different ticket, perhaps) what is happening here.

Reply:
ok i removed this change as a closer look tells me that i dont need to close files